bigbite/themer
# [BBT-106] Add clear all customisations & move export button into more menu dropdown

## Description



Completes [BBT-106](https://b5ecom.atlassian.net/browse/BBT-106) - This builds on the changes made in BBT-83 (I've branched off that branch), and re-adds the clear all customisations functionality back but this time into an `MoreMenuDropdown` component in the theme topbar next to the save and reset button.

Currently when you click 'Clear all' you still have to click save to get that to be saved as the current config. I think this is good because it gives someone a 'second chance' so to speak if they accidentally click it. However, I can change this if you think it's bad UX to have to click save again after clearing.

I noticed the image on this ticket was showing the export button being in here too, so I've moved that into this menu from below the preview.

## Change Log



- Added new `<MoreMenuDropdown>` next to the save and reset buttons that contains 'Clear all customisations' and 'Export theme.json' buttons.

## Steps to test



- Make sure you have some styles set in your theme.json
- Load Themer and make some customisations and click save.
- Make some more customisations and click reset. This should reset back to the previously saved customisations.
- Open the menu dropdown and click clear all customisations. This should reset back to the original theme.json.
- Make some customisations, click save and then ensure the export button still works as expected.

## Screenshots/Videos



### Interface

<img width="1568" alt="Screenshot 2023-10-27 at 11 06 19" src="https://github.com/bigbite/themer/assets/45216035/7427b995-5b1d-4cb2-949a-d998082f7dff">

### Clear all customisations

http://bigbite.im/v/vDeI7h

## Checklist:





[BBT-106]: https://b5ecom.atlassian.net/browse/BBT-106?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
